CONTRIBUTING.md

```diff
@@ -1,123 +1,126 @@
+welcome contributions from everyone.  Below are various bits of information to
+help you get started.  If you require additional help, please reach out to us on
+our [zulip channel](https://near.zulipchat.com/).
 
-We welcome all external contributions. This document outlines the process of contributing to nearcore.
-For contributing to other repositories, see `CONTRIBUTING.md` in the corresponding repository.
-For non-technical contributions, such as e.g. content or events, see [this document](https://docs.nearprotocol.com/docs/contribution/contribution-overview).
+# Getting started
 
-For an overview of the NEAR core architecture, see [this playlist](https://www.youtube.com/playlist?list=PL9tzQn_TEuFV4qlts0tVgndnytFs4QSYo).
+- For non-technical contributions, such as e.g. content or events, see [this
+document](https://docs.nearprotocol.com/docs/contribution/contribution-overview).
 
-# Pull Requests and Issues
+- For an overview of the NEAR core architecture, see [this
+playlist](https://www.youtube.com/playlist?list=PL9tzQn_TEuFV4qlts0tVgndnytFs4QSYo).
 
-All the contributions to `nearcore` happen via Pull Requests. To create a Pull Request, fork `nearcore`, create a new branch, do the work there, and then send the PR via the Github interface.
+- If you have never contributed to nearcore before and are looking for
+relatively simple issues to get started with, please checkout issues labeled
+with `C-good-first-issue`
+[here](https://github.com/near/nearcore/labels/C-good-first-issue).  If you see
+one that looks interesting and is not claimed, please comment on the issue that
+you would like to start working on it and someone from the team should assign it
+to you.
 
-The PRs should always be against the `master` branch.
+- If you have an idea for an enhancement to the protocol, please make the
+proposal by following the [NEAR Enhancement
+Proposal](https://github.com/near/NEPs/blob/master/neps/nep-0001.md).
 
-The exact process depends on the particular contribution you are making.
+- Otherwise, if you have an idea for an enhancement that does not require a
+change in the proposal, please start by creaeting a new issue in the repository.
+Someone should respond within 48 hours.  If the proposal is accepted, then you
+can follow the process below to begin working on it.
 
-## Typos or small fixes
+- Finally, if you have noticed an obvious typo or bug that can be fixed with a
+small code change, please feel free to submit a PR directly addressing the issue
+without opening an issue.
 
-If you see an obvious typo, or an obvious bug that can be fixed with a small change, in the code or documentation, feel free to submit the pull request that fixes it without opening an issue.
+If you have an issue assigned to you, then we kindly ask that you provide an
+update at least once a week so we know if you are still actively working on the
+issue.
 
-## Working on current tasks
+# Setting up editors and IDEs
```

Comment:
Extreme HTML pedantry, but `#` generates an `h1`, and there should be only one `h1` in an article. You want to use `##` for section titeles, and resreve `#` for overall titele of the document.